packages/neos-ui-editors/src/CodeMirror/CodeMirrorWrap/index.js

```diff
@@ -0,0 +1,42 @@
+import React, {PureComponent, PropTypes} from 'react';
+import CodeMirror from 'react-codemirror';
+
+// TODO: Find way to dynamically load any mode?
+import XmlMode from 'codemirror/mode/xml/xml';
+import HtmlMixedMode from 'codemirror/mode/htmlmixed/htmlmixed';
+import JavascriptMode from 'codemirror/mode/javascript/javascript';
+import HandlebarsMode from 'codemirror/mode/handlebars/handlebars';
+import MarkdownMode from 'codemirror/mode/markdown/markdown';
+import YamlMode from 'codemirror/mode/yaml/yaml';
+
+export default class CodeMirrorWrap extends PureComponent {
+    static propTypes = {
+        onChange: PropTypes.func.isRequired,
+        highlightingMode: PropTypes.string.isRequired,
+        value: PropTypes.string
+    };
+
+    constructor(props) {
+        super(props);
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(newValue) {
```

Comment:
I usually write these "handle..." methods *after* render(), as this reads better for me...